### PR TITLE
31/10 commit

### PR DESCRIPTION
**Pull request by Watxesnavo**

### CODE REVIEW

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    private final Bulldozer bulldozer1 = new Bulldozer();
+    private final Bulldozer bulldozer2 = new Bulldozer();
+    private final Bulldozer bulldozer3 = new Bulldozer();
```
Comment (zachar-chorny) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:9]: You don't have to do it that way. Create new objects inside the get method

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    private final Excavator excavator1 = new Excavator();
```
Comment (zachar-chorny) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:7]: same

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 3]
```diff
@@ -1,7 +1,45 @@
...
+            List<? extends Machine> machines = new TruckProducer().get();
+            return new ArrayList<>(machines);
+        }
+        return new ArrayList<>();
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        int size = machines.size();
+        for (int i = 0; i < size; i++) {
```
Comment (zachar-chorny) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java:34]:
```suggestion
        for (int i = 0; i < machines.size(); i++) {
```

**Review by zachar-chorny — CHANGES_REQUESTED** [review]

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    @Override
+    public List<Bulldozer> get() {
+        final Bulldozer bulldozer1 = new Bulldozer();
```
Comment (kateryna-deshchinska) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:9]:
```suggestion
        Bulldozer firstBulldozer = new Bulldozer();
```
change name like this

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+import core.mate.academy.model.Excavator;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    @Override
+    public List<Excavator> get() {
+        final Excavator excavator1 = new Excavator();
+        final Excavator excavator2 = new Excavator();
+        final Excavator excavator3 = new Excavator();
```
Comment (kateryna-deshchinska) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:11]: change name
and why do you use final?

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 6]
```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<T> {
```
Comment (kateryna-deshchinska) [inline src/main/java/core/mate/academy/service/MachineService.java:9]:
```suggestion
public interface MachineService<T extends Machine> {
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 7]
```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```
Comment (kateryna-deshchinska) [inline src/main/java/core/mate/academy/service/MachineService.java:20]:
```suggestion
    List<T> getAll(Class<? extends T> type);
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 8]
```diff
@@ -26,13 +27,13 @@ public interface MachineService {
      * @param machines - list of machines to be filled with value
      * @param value    - any object of machine sub class
      */
-    void fill(List<Object> machines, Object value);
+    void fill(List<? super Machine> machines, T value);
```
Comment (kateryna-deshchinska) [inline src/main/java/core/mate/academy/service/MachineService.java:30]:
```suggestion
    void fill(List<? super T> machines, T value);
```

File: src/main/java/core/mate/academy/service/MachineService.java [hunk 9]
```diff
@@ -26,13 +27,13 @@ public interface MachineService {
...
 
     /**
      * Call the method doWork() from each machine.
      * This method should be able to accept a list of bulldozers as well as list of trucks.
      *
      * @param machines - the list of machines
      */
-    void startWorking(List<Object> machines);
+    void startWorking(List<? extends Machine> machines);
```
Comment (kateryna-deshchinska) [inline src/main/java/core/mate/academy/service/MachineService.java:38]:
```suggestion
    void startWorking(List<? extends T> machines);
```

File: src/main/java/core/mate/academy/service/TruckProducer.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    @Override
+    public List<Truck> get() {
+        final Truck truck1 = new Truck();
+        final Truck truck2 = new Truck();
+        final Truck truck3 = new Truck();
```
Comment (kateryna-deshchinska) [inline src/main/java/core/mate/academy/service/TruckProducer.java:11]: same

**Review by kateryna-deshchinska — CHANGES_REQUESTED** [review]

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**